addons/mail/static/tests/helpers/mock_server/controllers/discuss.js

```diff
@@ -29,4 +41,68 @@ patch(MockServer.prototype, 'mail/controllers/discuss', {
         const [memberOfCurrentUser] = this.getRecords('mail.channel.member', [['channel_id', '=', channel_id], ['partner_id', '=', partnerId]]);
         this._mockMailChannelMember_NotifyTyping([memberOfCurrentUser.id], is_typing);
     },
+    /**
+     * Simulates the `/mail/rtc/channel/join_call` route.
+     *
+     * @private
+     * @param {integer} channel_id
+     * @returns {integer[]} [check_rtc_session_ids]
+     */
+    async _mockRouteMailRtcChannelJoinCall(channel_id, check_rtc_session_ids = []) {
+        const [currentChannelMember] = this.getRecords('mail.channel.member', [
+            ['channel_id', '=', channel_id],
+            ['partner_id', '=', this.currentPartnerId],
+        ]);
+        const sessionId = this.pyEnv['mail.channel.rtc.session'].create({
+            channel_member_id: currentChannelMember.id,
+            channel_id, // on the server, this is a related field from channel_member_id and not explicitly set
+        });
+        const channelMembers = this.getRecords('mail.channel.member', [['channel_id', '=', channel_id]]);
+        const rtcSessions = this.getRecords('mail.channel.rtc.session', [
+            ['channel_member_id', 'in', channelMembers.map(channelMember => channelMember.id)],
+        ]);
+        return {
+            'iceServers': false,
+            'rtcSessions': [
+                ['insert', rtcSessions.map(rtcSession => this._mockMailChannelRtcSession_MailChannelRtcSessionFormat(rtcSession.id))],
+            ],
+            'sessionId': sessionId,
+        };
+    },
+    /**
+     * Simulates the `/mail/rtc/channel/leave_call` route.
+     *
+     * @private
+     * @param {integer} channelId
+     */
+    async _mockRouteMailRtcChannelLeaveCall(channel_id) {
+        const channelMembers = this.getRecords('mail.channel.member', [['channel_id', '=', channel_id]]);
+        const rtcSessions = this.getRecords('mail.channel.rtc.session', [
+            ['channel_member_id', 'in', channelMembers.map(channelMember => channelMember.id)],
+        ]);
+        const notifications = [];
+        const channelInfo = this._mockMailChannelRtcSession_MailChannelRtcSessionFormatByChannel(rtcSessions.map(rtcSession => rtcSession.id));
+        for (const [channelId, sessionsData] of Object.entries(channelInfo)) {
+            const notificationRtcSessions = sessionsData.map((sessionsDataPoint) => {
+                return { 'id': sessionsDataPoint.id };
+            });
+            notifications.push([
+                channelId,
+                'mail.channel/rtc_sessions_update',
+                {
+                    'id': +channelId, // JS object keys are strings, but the type from the server is number
```

Comment:
properly call the parser function. There hidden type casting are hard to see at a glance